codebundles/k8s-deployment-healthcheck/deployment_logs.sh

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+
+# -----------------------------------------------------------------------------
+# Script Information and Metadata
+# -----------------------------------------------------------------------------
+# Author: @stewartshea
+# Description: This script is designed to fetch and process Kubernetes logs 
+# and provide helpful insights based on the logs. It uses lnav to sift and 
+# query the logs for detail, and then tryies to match namespace resources with 
+# some of the text created by the queries. This can be extended as needed to 
+# cover many logfile use cases
+# -----------------------------------------------------------------------------
+
+# Update PATH to ensure script dependencies are found
+export PATH="$PATH:$HOME/.lnav:$HOME/.local/bin"
+
+# -------------------------- Function Definitions -----------------------------
+
+# Check if a command exists
+function check_command_exists() {
+    if ! command -v $1 &> /dev/null; then
+        echo "$1 could not be found"
+        exit
+    fi
+}
+
+# ------------------------- Dependency Verification ---------------------------
+
+# Ensure all the required binaries are accessible
+check_command_exists ${KUBERNETES_DISTRIBUTION_BINARY}
+check_command_exists jq
+check_command_exists lnav
+
+# Load custom formats for lnav if it's installed
+# FIXME: This needs to sort out dev and runtime environment
+cp -rf /collection/extras/lnav/formats/* $HOME/.lnav/formats/installed
```

Comment:
Hmm... this is an interesting issue since there isn't path parity between dev / runtime. Normally I'd suggest placing the config in the codebundle dir but we'll be sharing this one around to many.

First thought is maybe we tack on some new ENVs for path information both in dev and runtime, but maybe there's a better way.

Reply:
Yeah, and this one specifically was a bit challenging with lnav... for some reason some of the docs specified certain areas where this could be loaded, or even a command line switch, but I struggled to get them to work consistently. The best I could do as $HOME/.lnav/formast/installed --